Source/Examples/ExampleLibrary/Axes/AxisExamples.cs

```diff
@@ -1409,5 +1346,81 @@ public static PlotModel MultiVerticalAxes()
+            return plotModel;
+        }
+
+        [Example("Manual Margins")]
+        public static PlotModel ManualMargins()
+        {
+            var plotModel = new PlotModel() { Title = "Manual Margins", Subtitle = "PlotMargins = 40", PlotMargins = new OxyThickness(40) };
+            plotModel.Axes.Add(new LinearAxis() { Position = AxisPosition.Bottom });
+            return plotModel;
+        }
+
+        [Example("Manual Left Margin")]
+        public static PlotModel ManualLeftMargin()
+        {
+            var plotModel = new PlotModel() { Title = "Manual Left Margin", Subtitle = "PlotMargins = 40,NaN,NaN,NaN", PlotMargins = new OxyThickness(40, double.NaN, double.NaN, double.NaN) };
+            plotModel.Axes.Add(new LinearAxis() { Position = AxisPosition.Bottom });
+            return plotModel;
+        }
+
+        [Example("Auto Margins - Wide Labels")]
+        public static PlotModel AutoMarginWideLabels()
+        {
+            var plotModel = new PlotModel() { Title = "Auto-adjusting plot margins - wide axis labels", Subtitle = "There should be enough space reserved such that the axis labels always fit in the viewport" };
+            plotModel.Axes.Add(GetLongLabelSeries());
+            return plotModel;
+        }
+
+        [Example("Auto Margins - Wide Labels, rotated")]
+        public static PlotModel AutoMarginWideLabelsRotated()
+        {
+            var plotModel = new PlotModel() { Title = "Auto-adjusting plot margins - wide rotated axis labels", Subtitle = "There should be enough space reserved such that the axis labels always fit in the viewport" };
+            var axis = GetLongLabelSeries();
+            axis.Angle = -90;
+            plotModel.Axes.Add(axis);
+            return plotModel;
+        }
+
+        [Example("Auto Margins - Wide Labels, fixed Range")]
+        public static PlotModel AutoMarginWideLabelsFixedRange()
+        {
+            var plotModel = new PlotModel() { Title = "Auto-adjusting plot margins - wide axis labels, fixed range", Subtitle = "When the axis range is fixed there should be no space reserved for axis labels" };
+            var axis = GetLongLabelSeries();
+            axis.IsPanEnabled = false;
+            axis.IsZoomEnabled = false;
+            plotModel.Axes.Add(axis);
+            return plotModel;
+        }
+
+        [Example("Auto Margins - Wide Labels, fixed Range 2")]
+        public static PlotModel AutoMarginWideLabelsFixedRange2()
+        {
+            var plotModel = new PlotModel() { Title = "Auto-adjusting plot margins - wide axis labels, fixed range", Subtitle = "The axis labels should exactly fit in the viewport" };
+            var axis = GetLongLabelSeries();
+            axis.IsPanEnabled = false;
+            axis.IsZoomEnabled = false;
+            axis.Minimum = -0.01;
+            axis.Maximum = 3.01;
+            plotModel.Axes.Add(axis);
+            return plotModel;
+        }
+
```

Comment:
Needs an example with reversed axis (these don't work at the moment):

        [Example("Auto Margins - Wide Labels, fixed Range 3 (reversed axis)")]
        public static PlotModel AutoMarginWideLabelsFixedRange3()
        {
            var plotModel = new PlotModel() { Title = "Auto-adjusting plot margins - wide axis labels, fixed range", Subtitle = "The axis labels should exactly fit in the viewport" };
            var axis = GetLongLabelSeries();
            axis.StartPosition = 1;
            axis.EndPosition = 0;
            axis.IsPanEnabled = false;
            axis.IsZoomEnabled = false;
            axis.Minimum = -0.01;
            axis.Maximum = 3.01;
            plotModel.Axes.Add(axis);
            return plotModel;
        }

Reply:
Oh wow you're absolutely right.
Misunderstanding from my side regarding what `ScreenMin` and `ScreenMax` actually mean...

Reply:
Maybe a 'reversed' checkbox to go with the 'transpose' checkbox would be a fun idea, so I can forget about it's existence as well.

Reply:
I've implemented a 'reversed' checkbox feature, and it didn't turn out as horribly as I thought it might, so I'll be putting a PR together soon.